### PR TITLE
Lh7070 support full backup

### PR DESCRIPTION
**Pull request by ChanYiLin**
- Add `ExecutionCount` to the recurringJob Status
- `BackupMode` to the `Backup` Spec

### CODE REVIEW

File: enhancements/20240314-recurring-and-manual-full-backup-support.md [hunk 1]
```diff
@@ -16,11 +16,19 @@ There is a chance that the previous backup blocks on the backupstore are corrupt
...
 ### Goals
 
-- Add a new fields `Parameters` to `RecurringJob` and `Backup`
-  - `backup-mode`: used in `Backup` CR to trigger the full backup (Options: `"full"`, `"incremental"`, default to `"incremental"` for always incremental)
-  - `full-backup-interval`: used in `RecurringJob - Backup Type` to execute full backup every N incremental backups (default to 0 for always incremental)
+- Add new fields to `RecurringJob`
+  - `Parameters` to `Spec`:
+    - `full-backup-interval`: used in `RecurringJob - Backup Type` to execute full backup every N incremental backups (default to 0 for always incremental)
+      - For example, if N is 5, then after 5 regular incremental backups, the job will perform full backup for the 6th backup
+  - `ExecutionCount` to `Status`:
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply enhancements/20240314-recurring-and-manual-full-backup-support.md:23]: Good question, I think yes
`ExecutionCount` should be neutral, it don't care if the job is successful or failed
So the every N times it executes the Full Backup make sense.
It shouldn't be every N successful times then executes Full Backup.
 
We can have `SuccessCount` and `FailedCount` in the future if needed.

Besides, if something happens to incremental backup and make it fail every time, then the full back job won't be executed anymore.

[the remark this answers is not in the document]
Comment (derekbit) [reply enhancements/20240314-recurring-and-manual-full-backup-support.md:23]: Got it. I'm fine with the current design. Thank you.

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
LGTM

**Pull request merged**